extensions/src/main/java/org/pentaho/platform/plugin/action/DefaultActionInvoker.java

```diff
@@ -124,8 +127,19 @@ public IActionInvokeStatus invokeAction( final IAction actionBean, final String
   protected IActionInvokeStatus invokeActionImpl( final IAction actionBean, final String actionUser, final
     Map<String, Serializable> params ) throws Exception {
 
+    final String workItemUid = WorkItemLifecycleEvent.getUidFromMap( params );
+    final WorkItemLifecycleEvent workItemLifecycleEvent = new WorkItemLifecycleEvent( workItemUid, StringUtil
+      .getMapAsPrettyString( params ) );
+
+    workItemLifecycleEvent.setWorkItemLifecyclePhase( WorkItemLifecyclePhase.IN_PROGRESS );
+    WorkItemLifecyclePublisher.publish( workItemLifecycleEvent );
+
     if ( actionBean == null || params == null ) {
-      throw new ActionInvocationException( Messages.getInstance().getCantInvokeNullAction() );
+      final String failureMessage = Messages.getInstance().getCantInvokeNullAction();
+      workItemLifecycleEvent.setWorkItemLifecyclePhase( WorkItemLifecyclePhase.FAILED );
+      workItemLifecycleEvent.setLifecycleDetails( failureMessage );
+      WorkItemLifecyclePublisher.publish( workItemLifecycleEvent );
```

Comment:
Suggestion: why not have a single publish method instead of to set things as part of separate calls.  Also it is clearer as to what should be publish by having it as part of method params list.